Added documentation about recommended Kubernetes labels (#9854)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/ref-kubernetes-labels.adoc

// Module included in the following assemblies:
//
// assembly-config.adoc

[id='ref-kubernetes-labels-{context}']
= Restrictions on Kubernetes labels

https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[Kubernetes labels] make it easier to organize, manage, and discover Kubernetes resources within your applications.
The Cluster Operator is responsible for applying the following Kubernetes labels to the operands it deploys.
These labels cannot be overridden through `template` configuration of Strimzi resources:

* `app.kubernetes.io/name`: Identifies the component type within Strimzi, such as `kafka`, `zookeeper`,  and`cruise-control`.
* `app.kubernetes.io/instance`: Represents the name of the custom resource to which the operand belongs to. For instance, if a Kafka custom resource is named `my-cluster`, this label will bear that name on the associated pods.
* `app.kubernetes.io/part-of`: Similar to `app.kubernetes.io/instance`, but prefixed with `strimzi-`.
* `app.kubernetes.io/managed-by`: Defines the application responsible for managing the operand, such as `strimzi-cluster-operator` or `strimzi-user-operator`.

> .Example Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`

Reviewer: ```suggestion
.Example Kubernetes labels on a Kafka pod when deploying a `Kafka` custom resource named `my-cluster`
```

// Module included in the following assemblies:
//
// assembly-config.adoc

[id='ref-kubernetes-labels-{context}']

> = Kubernetes labels

Reviewer: Not sure about this title. I wanted something active and generally about labels, but also imply some restrictions. Might be okay

> = Kubernetes labels

Author: Isn't the title misleading by adding "Configuring"? It seems to suggest you can configure them while it's the opposite. Of course, I am not saying that "Kubernetes labels" is the best title, maybe there is something even better?

> = Kubernetes labels

Author: Yeah I like this title more. Thanks!

> The Kubernetes documentation defines some https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[recommended labels] that could be set on the resources deployed within your application, in order to help describing objects in a way that all tools can understand.
> The Strimzi operator applies some of them to the operands it deploys:

Reviewer: ```suggestion
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[Kubernetes labels] make it easier to organize, manage, and discover Kubernetes resources within your applications.
You can configure your own labels through the `template` configuration of Strimzi resources.
However, the Cluster Operator is responsible for applying the following Kubernetes labels to the operands it deploys. These labels cannot be overridden through configuration:
```

> * `app.kubernetes.io/name`: it is the component type within Strimzi which could be `kafka`, `zookeeper`, `cruise-control`, and so on.
> * `app.kubernetes.io/instance`: it is the name of the custom resource to which the operand belong to (that is, `my-cluster` as the name of the `Kafka` custom resource applied on the Kafka pods).
> * `app.kubernetes.io/part-of`: it is the same as the `app.kubernetes.io/instance` but with a `strimzi-` prefix.
> * `app.kubernetes.io/managed-by`: it defines which application is in charge of managing the operand like `strimzi-cluster-operator`, `strimzi-user-operator` and so on.

Reviewer: ```suggestion
* `app.kubernetes.io/name`: Identifies the component type within Strimzi, such as `kafka`, `zookeeper`,  and`cruise-control`.
* `app.kubernetes.io/instance`: Represents the name of the custom resource to which the operand belongs to. For instance, if a Kafka custom resource is named `my-cluster`, this label will bear that name on the associated pods.
* `app.kubernetes.io/part-of`: Similar to `app.kubernetes.io/instance`, but prefixed with `strimzi-`.
* `app.kubernetes.io/managed-by`: Defines the application responsible for managing the operand, such as `strimzi-cluster-operator` or `strimzi-user-operator`.
```

> .Example of Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`

Reviewer: ```suggestion
.Example Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`
```

[source,yaml,subs=attributes+]
----
apiVersion: {KafkaApiVersion}
kind: Pod
metadata:
  name: my-cluster-kafka-0
  labels:
    app.kubernetes.io/instance: my-cluster
    app.kubernetes.io/managed-by: strimzi-cluster-operator
    app.kubernetes.io/name: kafka
    app.kubernetes.io/part-of: strimzi-my-cluster
spec:
  # ...
----

> Those labels cannot be overridden by the user (that is, by using templates).

Reviewer: ```suggestion

```
